heft-plugins/heft-sass-plugin/src/SassTypingsGenerator.ts

```diff
@@ -95,25 +110,53 @@ export class SassTypingsGenerator extends StringValuesTypingsGenerator {
       fileExtensions,
       filesToIgnore: sassConfiguration.excludeFiles,
 
+      getAdditionalOutputFiles: getCssPaths,
+
       // Generate typings function
-      parseAndGenerateTypings: async (fileContents: string, filePath: string) => {
+      parseAndGenerateTypings: async (fileContents: string, filePath: string, relativePath: string) => {
         if (this._isSassPartial(filePath)) {
           // Do not generate typings for Sass partials.
           return;
         }
+
         const css: string = await this._transpileSassAsync(
           fileContents,
           filePath,
           buildFolder,
           sassConfiguration.importIncludePaths
         );
-        const classNames: string[] = await this._getClassNamesFromCSSAsync(css, filePath);
-        const sortedClassNames: string[] = classNames.sort((a, b) => a.localeCompare(b));
-        const sassTypings: IStringValueTypings = { typings: [] };
-        for (const exportName of sortedClassNames) {
-          sassTypings.typings.push({ exportName });
+
+        let classMap: IClassMap = {};
+        const cssModulesClassMapPlugin: postcss.Plugin<TCssModules> = cssModules({
+          getJSON: (cssFileName: string, json: IClassMap) => {
+            classMap = json;
+          },
+          // Avoid unnecessary name hashing.
+          generateScopedName: (name: string) => name
```

Comment:
These are synchronous callbacks?

Reply:
Guaranteed to be invoked by the time the awaited promise completes.